core/state_processor.go

```diff
@@ -124,6 +124,16 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, gp *GasPool
 		return nil, ErrUnprotectedTransaction
 	}
 
+	// TODO(paul): Add CIP number
+	// Since CIP XXX full node incentives are deprecated.
+	// Check that neither `GatewayFeeRecipient` nor `GatewayFee` are set, otherwise reject the transaction
+	if config.IsGFork(blockNumber) {
+		gatewayFeeSet := !(msg.GatewayFee() == nil || msg.GatewayFee().Cmp(common.Big0) == 0)
```

Comment:
So, we have txs, that maybe have the gatewayFee in zero?

Reply:
Yes, that's what I found while adapting the tests.